lib/resources/dstu2/appointment.yaml

```diff
@@ -15,8 +27,99 @@ fields:
                    "display": "Community medicine",
                    "userSelected": false}],
                 "text": "Same Day"}
+  url: http://hl7.org/fhir/DSTU2/appointment-definitions.html#Appointment.type
   binding:
     description:  Additional details about where the content was created (e.g. clinical specialty).
     terminology:
     - display: Practice Setting Code Value Set
       system: http://hl7.org/fhir/ValueSet/c80-practice-codes
+
+- name: slot
+  required: 'No'
+  cardinality: 0..1
+  type: Reference (Slot)
+  description: The slot that this appointment is filling. If provided, then no schedule and start/end values MUST match the ones from the slot.
+  example: |
+    {
+      "slot": {
+        "reference": "Slot/337947169-4048128-9537620-0"
+      }
+    }
+  url: http://hl7.org/fhir/DSTU2/appointment-definitions.html#Appointment.slot
+  action: create
+
+- name: status
+  required: 'Yes'
+  cardinality: 1..1
+  type: code
+  description: A code representing the overall status of the Appointment. Each of the participants has their own participation status which indicates their involvement in the process, however this status indicates the shared status.
+  note: When creating a new appointment, the status must be "proposed".
+  action: create
+  example: |
+      {
+        "status": "proposed"
+      }
+  url: http://hl7.org/fhir/DSTU2/appointment-definitions.html#Appointment.status
+
+- name: participant
+  required: 'Yes'
+  cardinality: 1..*
+  type: BackboneElement
+  description: 	Participants involved in appointment.
+  note: When creating a new appointment, at most one participant can be populated.
+  action: create
+  example: |
+      {
+        "participant": [
+          {
+            "actor": {
+              "reference": "Patient/3886413",
+              "display": "STRECKER, TESTA"
+            },
+            "required": "required",
+            "status": "needs-action"
+          }
+        ]
+      }
+  children:
+
+  - name: type
+    required: 'No'
+    cardinality: 1..*
+    type: CodeableConcept
+    description: Role of participant in the appointment
+    note: When creating a new appointment, type cannot be populated.
+
+  - name: actor
+    required: 'Yes'
+    cardinality: 0..1
+    type: Reference (Patient)
+    description: Free text dosage information as reported about a patient's medication use.
+    note: When creating a new appointment, this must be a Patient participant.
+    example: |
+      "actor": {
+              "reference": "Patient/3886413",
+              "display": "Patient, Test"
+       }
+
+  - name: required
+    required: 'Yes'
+    cardinality: 0..1
+    type: code
+    description: Is this participant required to be present at the meeting.
+    note: When creating a new appointment, this value must be "required".
```

Comment:
For these and other fields - why require it and state what it must be set to? Why not allow it to be unset on creation and default it to the correct values behind the scenes? Would still need to fail unexpected values, but makes the API easier to consume.